bftengine/src/preprocessor/PreProcessor.hpp

```diff
@@ -19,10 +19,17 @@
 #include "SimpleThreadPool.hpp"
 #include "Replica.hpp"
 #include "RequestProcessingInfo.hpp"
+#include "sliver.hpp"
+#include "SigManager.hpp"
+
+#include <mutex>
 
 namespace preprocessor {
 
-class AsyncPreProcessJob;
+struct ClientRequestInfo {
+  std::recursive_mutex mutex_;  // Define a mutex per client to avoid contentions between clients
```

Comment:
nit - can this be simply `mutex` instead of `mutex_` ?

Reply:
Correct, in structures we use members without trailing underscore.